core/__tests__/actions/profiles.ts

```diff
@@ -369,31 +377,30 @@ describe("actions/profiles", () => {
           id: group.id,
           profileId: profile.id,
         };
-        const { error, success } = await specHelper.runAction(
-          "group:removeProfile",
-          connection
-        );
+        const { error, success } =
+          await specHelper.runAction<GroupRemoveProfile>(
+            "group:removeProfile",
+            connection
+          );
         expect(error).toBeUndefined();
         expect(success).toBeTruthy();
 
         connection.params = {
           csrfToken,
           id: profile.id,
         };
-        const { error: removeError, groups } = await specHelper.runAction(
-          "profile:view",
-          connection
-        );
+        const { error: removeError, groups } =
+          await specHelper.runAction<ProfileView>("profile:view", connection);
         expect(removeError).toBeUndefined();
         expect(groups.length).toBe(0);
       });
     });
 
     describe("search", () => {
-      let mario: Profile;
-      let luigi: Profile;
-      let peach: Profile;
-      let toad: Profile;
+      let mario: AsyncReturnType<ProfileCreate["run"]>["profile"];
```

Comment:
aha! we found some wrong types! 👍🏼

Reply:
Yeah! That's why I ended up with `AsyncReturnType<...` like you called out above.  But... using AppApiData is much cleaner! I'll update those.